api/src/Filter/ArchivedFilter.php

```diff
@@ -20,27 +21,27 @@ final class ArchivedFilter extends AbstractFilter implements GenericFilterInterf
      */
     public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null, array $context = []): void
     {
-        $archived = $context['filters'][self::PARAMETER_NAME] ?? null;
-        if ('true' !== $archived && $this->supports($resourceClass)) {
-            $queryBuilder->andWhere(sprintf('%s.archivedAt IS NULL', $queryBuilder->getRootAliases()[0] ?? 'o'));
+        if (!is_a($resourceClass, ArchivableInterface::class, true)) {
+            throw new \InvalidArgumentException("Can't apply the Archived filter on a resource ({$resourceClass}) not implementing the ArchivableInterface.");
         }
+
+        if ($this->normalizeValue($context['filters'][self::PARAMETER_NAME] ?? null)) {
+            return;
+        }
+
+        $queryBuilder->andWhere(sprintf('%s.archivedAt IS NULL', $queryBuilder->getRootAliases()[0] ?? 'o'));
     }
 
     /**
-     * {@inheritdoc}
+     * @param mixed $value
      */
-    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    private function normalizeValue($value): bool
```

Comment:
Private functions must be set after public ones

Reply:
OK, but it should be ensured by code analysis.